### ISSUE TITLE
Crash on changing a birthdate

### ISSUE THREAD

**Issue by Kreuger** [m-i-n-a-r/birday#111]
Hey. I have a birthday in my calendar that is wrong but whenever I try to change it, the app crashes after tapping update.

This is the crash report I got.
```

FATAL EXCEPTION: DefaultDispatcher-worker-1
Process: com.minar.birday, PID: 14647
android.database.sqlite.SQLiteConstraintException: UNIQUE constraint failed: Event.name, Event.surname, Event.originalDate (code 2067 SQLITE_CONSTRAINT_UNIQUE)
	at android.database.sqlite.SQLiteConnection.nativeExecuteForChangedRowCount(Native Method)
	at android.database.sqlite.SQLiteConnection.executeForChangedRowCount(SQLiteConnection.java:890)
	at android.database.sqlite.SQLiteSession.executeForChangedRowCount(SQLiteSession.java:756)
	at android.database.sqlite.SQLiteStatement.executeUpdateDelete(SQLiteStatement.java:66)
	at e.u.a.g.e.q(Unknown Source:2)
	at e.s.e.f(Unknown Source:7)
	at b.e.a.h.b.a(Unknown Source:12)
	at b.e.a.j.c$b.h(:2)
	at f.m.j.a.a.i(:2)
	at g.a.e0.run(Unknown Source:84)
	at g.a.s1.a.x(Unknown Source:0)
	at g.a.s1.a$a.run(:5)
```

**Comment by m-i-n-a-r** [comment]
Do you have another contact with the same name, surname and birthdate? Are you running version 2.1.0 or 3.x.x?

**Comment by Kreuger** [comment]
2.1.0. There was a duplicate that I didnt see. All good now!

**Comment by m-i-n-a-r** [comment]
Glad to hear that, but of course i'll have to handle this particular case, the app shouldn't simply crash. I'll take care of it in the next update!

**Comment by m-i-n-a-r** [comment]
Fixed in 7ad76953edd4f84293d43cbf62dec625db40650f , it was a missing conflict strategy. Now the app will overwrite the duplicate instead of crash 😄

**Comment by Kreuger** [comment]
That was quick. Well done!
